R/remove_funs.R

```diff
@@ -22,12 +22,22 @@ crop <- function ( x , char = " " ) {
 
 ### gsub for more than one pattern
 gsubAll <- function(string, old, new) {
-  stopifnot ( is.character(string), is.character(old), is.character(new), length(old) == length(new))
+  checkmate::assert_character(string)
+  lapply(list(old, new), FUN = checkmate::assert_character,unique=TRUE, any.missing = FALSE, min.len = 2)
+  stopifnot ( length(old) == length(new))
+  stopifnot (length(intersect(old, new))==0)                                    ### in einem Zwischenschritt unique IDs mit gleich vielen characters erzeugen
+  inter <- stringi::stri_rand_strings(n=length(old), length=12, pattern="[A-Za-z0-9]")
+  stopifnot(length(unique(inter)) == length(inter))
+  ind <- sort(nchar(old), decreasing = TRUE, index.return=TRUE)                 ### sortieren nach nchar(), wenn kuerzere strings in laengeren enthalten
```

Comment:
The comment reads `wenn kuerzere strings in laengeren enthalten`, but the action is performed regardless? Maybe the comment could be modified to reflect that this action is always performed?

Reply:
Well, the comment is sloppy. What is meant: 
`sort by nchar() to prevent recursive substitution if shorter strings are contained in longer strings`